### ISSUE TITLE
.NET Core 3 Preview 8 WPF Build error on custom BaseIntermediateOutputPath 

### ISSUE THREAD

**Issue by TFTomSun** [dotnet/wpf#1718]
* .NET Core Version: 3.0 Preview 8
* Windows version: Windows 10
* Does the bug reproduce also in WPF for .NET Framework 4.8?: No

 
 **Problem description:**
 I set the BaseIntermediateOutputPath to >$(OutputRootDirectory)obj\$(Configuration)\$(MSBuildProjectName)\
where OutputRootDirectory is a shared Output path of many projects
It is set via a build sdk that is included in the Directory.Build.props and Directory.Build.targets files in the root of my repository.

 **Actual behavior:** 
 When I build the project I get alot of build errors for xaml files, for example:
>D:\Git\Collaboration\Net\Framework\_out\obj\Debug\Siemens.Collaboration.Net.Presentation\netcoreapp3.0\Controls\AvalonDockApplicationControl.g.cs(54,18,54,62): error CS1504: Source file 'Controls\AvalonDockApplicationControl.xaml' could not be opened -- Could not find file.

It seems like the line information generated in to the g.cs files don't honor the the changed BaseIntermediateOutputPath setting.

 **Expected behavior:**
The build just runs fine

**Comment by vatsan-madhavan** [comment]
Likely related to or duplicate of https://github.com/dotnet/wpf/issues/366

**Comment by grubioe** [comment]
Closing as duplicate of #366

**Comment by TFTomSun** [comment]
@grubioe I'm not sure whether this is a duplicate. It's more about how the compiler resolves line preprocessor statements with relative paths. For example into one of the generated cs files in the intermediate ouput is the following line statement generated:

>  #line 16 "Controls\AvalonDockApplicationControl.xaml"

I don't know how the compiler can find the path to the xaml file, because the xaml file is in my case not relatively located to the base intermediate output. I'm afraid that there's somewhere an assumption that the intermediate output is below the project directory.

**Comment by TFTomSun** [comment]
I think I found the issue. The implementation of ParentFolderPath in the MarkupCompiler class of the PresentationBuildTasks assembly seems not to be clean:

```cs 
private string ParentFolderPrefix
    {
      get
      {
        string empty = string.Empty;
        if (this.TargetPath.StartsWith(this.SourceFileInfo.SourcePath, StringComparison.OrdinalIgnoreCase))
        {
          string[] strArray = (this.TargetPath.Substring(this.SourceFileInfo.SourcePath.Length) + this.SourceFileInfo.RelativeSourceFilePath).Split('\\');
          for (int index = 1; index < strArray.Length; ++index)
            empty += "..\\";
        }
        return empty;
      }
    }

```

**Comment by wjk** [comment]
@TFTomSun Open source is cool! No need to decompile anymore: 

https://github.com/dotnet/wpf/blob/55815c9287682c1621cd5f89ffdd75ae770f4ea6/src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/MarkupCompiler/MarkupCompiler.cs#L1567-L1585

**Comment by TFTomSun** [comment]
**off topic**
@wjk maybe I don't know a technique yet how to browse and search for symbols with the original sources. What I did:

* check what the targets did in the PresentationBuildTasks package and found the task
* opened the assembly with dot peek
* navigated to the task
* drilled down to the methods via navigate to/ find symbols

Is there a similar way to do that based on the original sources? Finding and cloning the whole repository seems to be more complicated for me

**Comment by wjk** [comment]
@TFTomSun If you know what class name you're looking for, you can go to the homepage for this repo, press the "T" key on your keyboard, and type in the class name into the File Finder page that appears. Since most filenames in this repo share names with the classes they contain, this is usually a good way to find one class from the (admittedly large) repository.

Failing that, the [src directory](https://github.com/dotnet/wpf/tree/master/src/Microsoft.DotNet.Wpf/src) contains all the source code for the entire WPF project, organized in subfolders by assembly, then by namespace. If you're using PowerShell, you can use the following command to find a string out of all the C# source files in the repo:

```powershell
cd path_to_wpf_repo_clone\src\Microsoft.DotNet.Wpf\src
dir -Recurse . -Include *.cs | sls "your_string_here" -SimpleMatch
```

Or, if you prefer working on a Linux or Mac box, I would recommend using the [ack program](https://beyondgrep.com/) instead. All you need to do to use it is type `ack some.regex.here`, and it will search every file under the current directory for that regex, and print filenames and line numbers for each match it finds -- and it's fast. Hope this helps!

**Comment by TFTomSun** [comment]
@wjk ok, i thought there might be a more feasible way to navigate the sources...  i prefer a symbol based navigation support via decompilation until we can do that via visual code directly online one day.

**Comment by weltkante** [comment]
@TFTomSun something like reference source but for .NET Core repositories would be great. Roslyn has one I believe, not sure about CoreFX. I'm still using the Desktop Reference Source for navigating the sources but they are moving apart quickly, in particular WinForms is doing heavy refactoring.

**Comment by TFTomSun** [comment]
@weltkante yes, that looks promising. I hope something like this will be integrated into github one day, so that you can use it for any project.

**Comment by weltkante** [comment]
@TFTomSun apparently WinForms and the managed part of WPF are already indexed in https://source.dot.net/

**Comment by tmat** [comment]
The fix would be to use full paths to `xaml` files in`#line` and `#checksum` directives of `.g.cs` files.

**Comment by epvanhouten** [comment]
I ran into this issue while adding packaging to a dotNet Framework based solution. Working around by narrowing who gets the 'EmbedUntrackedSources' property using a targets file.

**Comment by KirillOsenkov** [comment]
Related: https://github.com/dotnet/sdk/issues/34438
